File: nom-api/Nom.Orch/Models/Cookbook/CookbookResponseModel.cs
namespace Nom.Orch.Models.Cookbook
{
    public class CookbookResponseModel
    {
        public long Id { get; set; }
        public long HouseholdId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Slug { get; set; }
        public bool IsPublic { get; set; }
        public int RecipeCount { get; set; }
        public DateTime CreatedDate { get; set; }
    }
}

File: nom-ui/src/app/cookbook/models/cookbook-response.model.ts
export interface ICookbookResponseModel {
    id: number;
    householdId: number;
    name: string;
    description?: string;
    slug?: string;
    isPublic: boolean;
    recipeCount: number;
    createdDate: Date;
}

export class CookbookResponseModel implements ICookbookResponseModel {
    id = 0;
    householdId = 0;
    name = '';
    description?: string;
    slug?: string;
    isPublic = false;
    recipeCount = 0;
    createdDate: Date = new Date();

    constructor(data?: Partial<ICookbookResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
}
